qwiic_i2c/__init__.py

```diff
@@ -63,11 +63,24 @@
 #
 #-----------------------------------------------------------------------------
 # Drivers and driver baseclass
-from .i2c_driver 	import I2CDriver
-from .linux_i2c 	import LinuxI2C
-from .circuitpy_i2c import CircuitPythonI2C
+from .i2c_driver import I2CDriver
 
-_drivers = [LinuxI2C, CircuitPythonI2C]
+# All supported platform module and class names
+sub_modules = ["linux_i2c", "circuitpy_i2c", "micropython_i2c"]
+class_names = ["LinuxI2C", "CircuitPythonI2C", "MicroPythonI2C"]
+
+# List of platform drivers found on this system
+_drivers = []
+
+# Loop through all supported platform drivers, see if they're included, and
+# append them to the driver list if so
+for i in range(len(sub_modules)):
+	try:
+		full_name = "qwiic_i2c." + sub_modules[i]
```

Comment:
You could skip the creation of the full_name variable and just put the RHS of this statement in the call to import.